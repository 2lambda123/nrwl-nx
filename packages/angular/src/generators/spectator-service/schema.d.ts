export interface Schema {
  name: string;
  project: string;
  path?: string;
  skipTests?: boolean;
  jest?: boolean;
}

export interface NormalizedSchema extends Schema {
  directory: string;
  filePath: string;
  projectSourceRoot: string;
  projectRoot: string;
}
