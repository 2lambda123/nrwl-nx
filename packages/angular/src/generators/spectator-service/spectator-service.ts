import type { Tree } from '@nx/devkit';
import { joinPathFragments, names, generateFiles } from '@nx/devkit';
import { normalizeOptions, validateOptions } from './lib';
import type { Schema } from './schema';

export async function spectatorServiceGenerator(
  tree: Tree,
  rawOptions: Schema
) {
  validateOptions(tree, rawOptions);
  const options = normalizeOptions(tree, rawOptions);

  const serviceNames = names(options.name);

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.directory,
    {
      fileName: serviceNames.fileName,
      className: serviceNames.className,
      jest: options.jest,
      tpl: '',
    }
  );

  if (options.skipTests) {
    const pathToSpecFile = joinPathFragments(
      options.directory,
      `${serviceNames.fileName}.service.spec.ts`
    );

    tree.delete(pathToSpecFile);
  }
}

export default spectatorServiceGenerator;
