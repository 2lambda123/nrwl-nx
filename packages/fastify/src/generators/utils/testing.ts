import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export function createTreeWithFastifyApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  tree.write(
    'workspace.json',
    String.raw`
      {
        "projects": {
          "${appName}": {
            "root": "apps/api",
            "sourceRoot": "apps/api/src",
            "projectType": "application",
            "prefix": "api",
            "targets":{}
          }
        }
      }
    `
  );

  return tree;
}
