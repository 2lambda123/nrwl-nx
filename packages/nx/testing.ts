export {cold, hot} from 'jasmine-marbles';
export {readAll} from './src/testing-utils';
