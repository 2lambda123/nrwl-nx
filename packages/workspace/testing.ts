export { MockBuilderContext } from './src/utils/testing-utils';
