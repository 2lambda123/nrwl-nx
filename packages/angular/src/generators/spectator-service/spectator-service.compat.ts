import { convertNxGenerator } from '@nx/devkit';
import { warnForSchematicUsage } from '../utils/warn-for-schematic-usage';
import { spectatorServiceGenerator } from './spectator-service';

export default warnForSchematicUsage(
  convertNxGenerator(spectatorServiceGenerator)
);
