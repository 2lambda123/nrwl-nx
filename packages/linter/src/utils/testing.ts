import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { Tree } from '@angular-devkit/schematics';

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/linter-migrations',
  join(__dirname, '../../migrations.json')
);

const testRunner = new SchematicTestRunner(
  '@nrwl/linter',
  join(__dirname, '../../collection.json')
);

export function runMigration<T = any>(
  migrationName: string,
  options: T,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync<T>(migrationName, options, tree)
    .toPromise();
}

export function runSchematic<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
