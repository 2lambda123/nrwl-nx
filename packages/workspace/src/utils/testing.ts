import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import type { Rule, Tree } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/workspace',
  join(__dirname, '../../generators.json')
);

testRunner.registerCollection(
  '@nrwl/jest',
  join(__dirname, '../../../jest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/cypress',
  join(__dirname, '../../../cypress/generators.json')
);

testRunner.registerCollection(
  '@nrwl/express',
  join(__dirname, '../../../express/generators.json')
);

testRunner.registerCollection(
  '@nrwl/react',
  join(__dirname, '../../../react/generators.json')
);

testRunner.registerCollection(
  '@nrwl/angular',
  join(__dirname, '../../../angular/generators.json')
);

testRunner.registerCollection(
  '@nrwl/next',
  join(__dirname, '../../../next/generators.json')
);

testRunner.registerCollection(
  '@nrwl/node',
  join(__dirname, '../../../node/generators.json')
);

testRunner.registerCollection(
  '@nrwl/nest',
  join(__dirname, '../../../nest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/web',
  join(__dirname, '../../../web/generators.json')
);

/**
 * @deprecated This will be removed in v17. Prefer writing Nx Generators with @nrwl/devkit. Generators can be tested by simply calling them in it().
 */
export function runSchematic<T extends object = any>(
  schematicName: string,
  options: T,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}

/**
 * @deprecated This will be removed in v17. Prefer writing Nx Generators with @nrwl/devkit. Generators can be tested by simply calling them in it().
 */
export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}
