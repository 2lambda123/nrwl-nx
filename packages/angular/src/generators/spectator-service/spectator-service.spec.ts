import { addProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { spectatorServiceGenerator } from './spectator-service';

describe('service Generator', () => {
  it('should create service files for library', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'lib1', {
      projectType: 'library',
      sourceRoot: 'libs/lib1/src',
      root: 'libs/lib1',
    });
    // ACT
    await spectatorServiceGenerator(tree, {
      name: 'services/example',
      project: 'lib1',
    });

    // ASSERT
    expect(
      tree.read('libs/lib1/src/lib/services/example.service.ts', 'utf-8')
    ).toMatchSnapshot('service');
    expect(
      tree.read('libs/lib1/src/lib/services/example.service.spec.ts', 'utf-8')
    ).toMatchSnapshot('service test file');
  });

  it('should create service files for application', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'app1', {
      projectType: 'application',
      sourceRoot: 'apps/app1/src',
      root: 'apps/app1',
    });

    // ACT
    await spectatorServiceGenerator(tree, {
      name: 'services/example',
      project: 'app1',
    });

    // ASSERT
    expect(
      tree.read('apps/app1/src/app/services/example.service.ts', 'utf-8')
    ).toMatchSnapshot('service');
    expect(
      tree.read('apps/app1/src/app/services/example.service.spec.ts', 'utf-8')
    ).toMatchSnapshot('service test file');
  });

  it('should create test files not using jest', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'lib1', {
      projectType: 'library',
      sourceRoot: 'libs/lib1/src',
      root: 'libs/lib1',
    });
    // ACT
    await spectatorServiceGenerator(tree, {
      name: 'services/example',
      project: 'lib1',
    });

    // ASSERT
    expect(
      tree.read('libs/lib1/src/lib/services/example.service.spec.ts', 'utf-8')
    ).not.toContain('jest');
  });

  it('should create test files using jest', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'lib1', {
      projectType: 'library',
      sourceRoot: 'libs/lib1/src',
      root: 'libs/lib1',
    });
    // ACT
    await spectatorServiceGenerator(tree, {
      name: 'services/example',
      project: 'lib1',
      jest: true,
    });

    // ASSERT
    expect(
      tree.read('libs/lib1/src/lib/services/example.service.spec.ts', 'utf-8')
    ).toContain('jest');
  });

  it('should not create test files', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addProjectConfiguration(tree, 'lib1', {
      projectType: 'library',
      sourceRoot: 'libs/lib1/src',
      root: 'libs/lib1',
    });
    // ACT
    await spectatorServiceGenerator(tree, {
      name: 'services/example',
      project: 'lib1',
      skipTests: true,
    });

    // ASSERT
    expect(
      tree.read('libs/lib1/src/services/example.service.ts', 'utf-8')
    ).toBeDefined();
    expect(
      tree.read('libs/lib1/src/services/example.service.spec.ts', 'utf-8')
    ).toEqual(null);
  });
});
