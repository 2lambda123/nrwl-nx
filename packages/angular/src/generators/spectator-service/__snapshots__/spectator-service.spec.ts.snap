// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service Generator should create service files for application: service 1`] = `
"import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ExampleService {

  constructor() {}
}"
`;

exports[`service Generator should create service files for application: service test file 1`] = `
"import { createServiceFactory, SpectatorService } from '@ngneat/spectator';
import { ExampleService } from './example.service';

describe('ExampleService', () => {
  let spectator: SpectatorService<ExampleService>;
  const createService = createServiceFactory(ExampleService);

  beforeEach(() => spectator = createService());

  it('should...', () => {
    expect(spectator.service).toBeTruthy();
  });
});"
`;

exports[`service Generator should create service files for library: service 1`] = `
"import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ExampleService {

  constructor() {}
}"
`;

exports[`service Generator should create service files for library: service test file 1`] = `
"import { createServiceFactory, SpectatorService } from '@ngneat/spectator';
import { ExampleService } from './example.service';

describe('ExampleService', () => {
  let spectator: SpectatorService<ExampleService>;
  const createService = createServiceFactory(ExampleService);

  beforeEach(() => spectator = createService());

  it('should...', () => {
    expect(spectator.service).toBeTruthy();
  });
});"
`;
